producer-kafka/src/main/scala/za/co/absa/spline/producer/kafka/ProducerKafkaConfig.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.producer.kafka
+
+import com.fasterxml.jackson.databind.{ObjectMapper, PropertyNamingStrategy}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.twitter.finatra.FinatraInternalModules
+import org.apache.commons.configuration.ConfigurationConverter
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.serialization.{ByteArrayDeserializer, StringDeserializer}
+import org.springframework.context.annotation.{Bean, ComponentScan, Configuration}
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
+import org.springframework.core.env.{MutablePropertySources, PropertiesPropertySource}
+import org.springframework.kafka.annotation.EnableKafka
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
+import org.springframework.kafka.core.{ConsumerFactory, DefaultKafkaConsumerFactory}
+import org.springframework.kafka.listener.ContainerProperties.AckMode
+import org.springframework.kafka.support.JacksonUtils
+import org.springframework.kafka.support.converter.Jackson2JavaTypeMapper.TypePrecedence
+import org.springframework.kafka.support.converter.{ByteArrayJsonMessageConverter, DefaultJackson2JavaTypeMapper, RecordMessageConverter}
+import za.co.absa.spline.common.config.DefaultConfigurationStack
+
+import java.util.concurrent.TimeUnit
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.{Duration, DurationInt}
+
+@EnableKafka
+@Configuration
+@ComponentScan(basePackageClasses = Array(classOf[listener._package]))
+class ProducerKafkaConfig {
+
+  import ProducerKafkaConfig._
+
+  @Bean
+  def kafkaListenerContainerFactory: ConcurrentKafkaListenerContainerFactory[_, _] = {
+    val factory = new ConcurrentKafkaListenerContainerFactory[String, AnyRef]()
+    factory.setConcurrency(1)
+    factory.setConsumerFactory(consumerFactory)
+    factory.setMessageConverter(messageConverter)
+    factory.getContainerProperties.setAckMode(AckMode.MANUAL)
+    factory
```

Comment:
Let me suggest a slightly different code style for initialization blocks like this one. It's not necessarily (or universally) better (creates an anonymous subclass, so it's a subject to watch out), but is IMO visually clearer. Just a hint.
```suggestion
    new ConcurrentKafkaListenerContainerFactory[String, AnyRef] {
      setConcurrency(1)
      setConsumerFactory(consumerFactory)
      setMessageConverter(messageConverter)
      getContainerProperties.setAckMode(AckMode.MANUAL)
    }
```